src/fam-api/fam.cpp

```diff
@@ -3861,9 +3861,26 @@ Fam_Descriptor *fam::fam_lookup(const char *itemName, const char *regionName) {
  */
 Fam_Region_Descriptor *
 fam::fam_create_region(const char *name, uint64_t size, mode_t permissions,
-                       Fam_Redundancy_Level redundancyLevel, ...) {
+                       Fam_Region_Attributes* regionAttributes) {
     TRY_CATCH_BEGIN
-    return pimpl_->fam_create_region(name, size, permissions, redundancyLevel);
+    if(regionAttributes == NULL)
```

Comment:
This "regionAttributes" parameter is passed by the application. 
Implementation should only read this parameter. 
If applicaiton call create_region api with "NULL", just use default parameters. 
We need to save these values in the region descriptor, by looking at the input regionAttribute. Either have separate copy of the regionAttribute structure or keep it as separate data items in the region descriptor.

Reply:
resolved